src/types/endpoints/service_endpoint.go

```diff
@@ -51,22 +51,26 @@ func (s *ServiceEndpoint) GetPortMappings() []string {
 func (s *ServiceEndpoint) GetEndpointMappings() map[string]string {
 	switch s.ServiceConfig.Type {
 	case "public":
-		return s.getExternalOrigin()
+		return s.getPublicEndpoints()
 	default:
 		return map[string]string{}
 	}
 }
 
-func (s *ServiceEndpoint) getExternalOrigin() map[string]string {
+func (s *ServiceEndpoint) getPublicEndpoints() map[string]string {
 	externalKey := fmt.Sprintf("%s_EXTERNAL_ORIGIN", toConstantCase(s.ServiceRole))
+	internalKey := fmt.Sprintf("%s_INTERNAL_ORIGIN", toConstantCase(s.ServiceRole))
+	endpoints := map[string]string{}
 	if s.BuildMode.Type == types.BuildModeTypeLocal {
 		if s.HostPort != "" {
```

Comment:
Unrelated to this PR but why do we check the hostPort is defined here? I don't think that is necessary to check at this point. Maybe in another PR we can set the hostPort in test environment and thus be able to remove this check

Reply:
Host port is only populated once a container port for the service has been attained. This was just to handle cases where the user hasn't defined a container port for a service yet to prevent passing env vars that looked like `https://localhost:`

Reply:
Then I think this should also prevent us from returning the env vars when building for remote. Could this become
```go
if s.ContainterPort != "" {
  if s.BuildMode.Type == types.BuildModeTypeLocal {
    // set local endpoint vars
  } else {
    // set remote endpoint vars
  }
}
```

Reply:
Changed it to that. Updated the constructor as a result